api/cloudcontroller/ccv3/job.go

```diff
@@ -114,6 +124,10 @@ func (client *Client) PollJob(jobURL JobURL) (Warnings, error) {
 			return allWarnings, nil
 		}
 
+		if job.IsAt(state) {
```

Comment:
Could also be:
```go
if job.State == state {
```
My personal opinion is that having a method makes the solution more complex.

Reply:
hmm, it is a fair point. I'd rather use a method though as the other conditions do it that way and I find it more readable when everything is at the same level of abstraction.